src/SharpBrick.PoweredUp/Functions/LinearSpeedChange.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace SharpBrick.PoweredUp.Functions
+{
+    public class LinearSpeedChange : IterativeChange<sbyte>
+    {
+        private readonly ILogger<LinearSpeedChange> _logger;
+        private TachoMotor _motor;
+        private sbyte _startSpeed;
+
+        public byte MaxPower { get; set; } = 100;
+
+        public sbyte IterationStep { get; private set; }
+
+        public LinearSpeedChange(ILogger<LinearSpeedChange> logger)
+            : base(logger)
+        {
+            _logger = logger;
+        }
+
+        public async Task ExecuteAsync(TachoMotor motor, sbyte startSpeed, sbyte endSpeed, int steps, int milliseconds, CancellationTokenSource cts = default)
```

Comment:
I am not so sure about this API interface. Should this routine auto-calculate the steps (to e.g. bring each change of motor speed in a sub (e.g.) 200ms interval?